spec.bs

```diff
@@ -451,25 +451,27 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
 1. Let |leadingBidInfo| be a new [=leading bid info=].
 1. Let |queue| be the result of [=starting a new parallel queue=].
 1. If |auctionConfig|'s [=auction config/component auctions=] are not [=list/is empty|empty=]:
-  1. [=Assert=] |isTopLevel| being true.
+  1. [=Assert=] |topLevelAuctionConfig| being null.
   1. Let |pendingComponentAuctions| be the [=list/size=] of |auctionConfig|'s
     [=auction config/component auctions=].
   1. [=list/For each=] |component| in |auctionConfig|'s [=auction config/component auctions=],
     [=parallel queue/enqueue steps|enqueue the following steps=] to |queue|:
-    1. Let |compWinner| be the result of [=generate and score bids=] with |component|, |global|, and
-      false.
-    1. If |compWinner| is not failure or null, [=score and rank a bid=] with |compWinner|,
-      |leadingBidInfo|, |desisionLogicScript|, and true.
+    1. Let |compWinner| be the result of [=generate and score bids=] with |component|,
+      |auctionConfig|, and |global|.
+    1. If |compWinner| is not failure or null, [=score and rank a bid=] with |auctionConfig|,
+      |compWinner|, |leadingBidInfo|, |desisionLogicScript|, and true.
     1. Decrement |pendingComponentAuctions| by 1.
   1. Wait until |pendingComponentAuctions| is 0.
   1. Return |leadingBidInfo|'s [=leading bid info/leading bid=].
 1. Let |allBuyersExperimentGroupId| be |auctionConfig|'s
   [=auction config/all buyer experiment group id=].
 1. Let |auctionSignals| be |auctionConfig|'s [=auction config/auction signals=].
-1. TODO: Let |browserSignals| be empty?
+1. Let |browserSignals| be an [=ordered map=] whose [=map/keys=] are [=strings=] and whose
+  [=map/values=] are {{any}}.
 1. TODO: Set |browserSignals|'s "topWindowHostname" to hostname of top window
-1. TODO: Set |browserSignals|'s "seller" to the seller
-1. TODO: Set |browserSignals|'s "topLevelSeller" to the top level seller
+1. [=map/Set=] |browserSignals|["seller"] to |auctionConfig|'s [=auction config/seller=].
+1. If |topLevelAuctionConfig| is not null:
+  1. [=map/Set=] |browserSignals|["topLevelSeller"] to |topLevelAuctionConfig|'s [=auction config/seller=].

```

Comment:
The map's value has a type of `{{any}}` which I think has to be a non-union WebIDL type https://webidl.spec.whatwg.org/#idl-any, and "origin" (the type of `config/seller`) is not a WebIDL type. Can you serialize this to a string? Is that what our implementation does?

Reply:
We serialize, so I made the spec serialize.